File: output/deepseek_r1/avatar/algo_based_translation/java/python/codeforces_456_A.py

n = int(input())
temp = 0
q = 0
w = 0
e = 0
f = False

for _ in range(n):
    x, y = map(int, input().split())
    
    if temp < x:
        if q > y:
            f = True
        q = y
        temp = x
    elif temp > x:
        if q < y:
            f = True
        q = y
        temp = x
    
    if x > w:
        if y < e:
            f = True
        w = x
        e = y
    if x < w:
        if y > e:
            f = True

print("Happy Alex" if f else "Poor Alex")

File: output/llama_4_maverick/avatar/algo_based_translation/python/java/codeforces_456_A.java

import java.util.Arrays;
import java.util.Scanner;

public class codeforces_456_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[][] m = new int[n][2];
        for (int i = 0; i < n; i++) {
            m[i][0] = scanner.nextInt();
            m[i][1] = scanner.nextInt();
        }
        Arrays.sort(m, (a, b) -> Integer.compare(b[0], a[0]));
        int[] a = m[0];
        boolean found = false;
        for (int i = 1; i < n; i++) {
            if (m[i][1] > a[1]) {
                System.out.println("Happy Alex");
                found = true;
                break;
            }
            a = m[i];
        }
        if (!found) {
            System.out.println("Poor Alex");
        }
    }
}

